lib/graphql/language/document_from_schema_definition.rb

```diff
@@ -18,18 +36,18 @@ def document
         )
       end
 
-      protected
+      def build_schema_node
+        schema_node = GraphQL::Language::Nodes::SchemaDefinition.new
 
-      def build_schema_node(schema)
-        schema_node = GraphQL::Language::Nodes::SchemaDefinition.new(
-          query: schema.query.name
-        )
+        if schema.query && warden.get_type(schema.query.name)
+          schema_node.query = schema.query.name
+        end
 
-        if schema.mutation
+        if schema.mutation && warden.get_type(schema.mutation.name)
           schema_node.mutation = schema.mutation.name
         end
 
-        if schema.subscription
+        if schema.subscription && warden.get_type(schema.subscription.name)
```

Comment:
How about using [`#root_type_for_operation("subscription")`](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/schema/warden.rb#L118) here and for `mutation` and `query` ?

Reply:
What's the difference here? It doesnt look like `root_type_for_operation` uses the warden 🤔

Reply:
Oh, to specify, `Warden#root_type_for_operation`, not `Schema#root_type_for_operation`: 

https://github.com/rmosolgo/graphql-ruby/blob/20d70eb374809835fedaf85bd8bffd0d71799abe/lib/graphql/schema/warden.rb#L118-L125

Reply:
ohhhhhh 💡 awesome.

Done :)